pkg/kubelet/cm/cpumanager/cpu_assignment.go

```diff
@@ -443,18 +520,21 @@ func (a *cpuAccumulator) takeFullSockets() {
 }
 
 func (a *cpuAccumulator) takeFullCores() {
-	for _, core := range a.freeCores() {
-		cpusInCore := a.topo.CPUDetails.CPUsInCores(core)
-		if !a.needsAtLeast(cpusInCore.Size()) {
-			continue
+	// only take full cores if we are using sortCPUsPacked
```

Comment:
I'd push this into `sortCPUsPackedOrSpread`. This method should delegate, much like we do in `freeCPUs`. This way: if we sort packed, we do as today; otherwise, the code turns into no-operation (NOP) - but let's make sure to add a comment to explain why. A couple lines are likely sufficient.

Reply:
sounds good. Let me update this snippet.

Reply:
@ffromani please have another check. I've also noticed that the current naming for our CPU allocation strategies might not be entirely intuitive. To enhance clarity and purpose, I suggest renaming the interface from sortCPUsPackedOrSpread to strategyCPUsPackedOrSpreadFuncs. do you think that make sense?

Current interface, putting`allocateFullCores()` is a little bit weird under `sortxxx`
```
type sortCPUsPackedOrSpreadFuncs interface {
	sortAvailableCores() []int
	allocateFullCores()
}
```

Reply:
the renaming you suggest makes sense. I'll have another (and final?) review later but I won't be against using this name.